app/src/main/java/com/humberto/tasky/core/database/dao/AgendaDao.kt

```diff
@@ -0,0 +1,24 @@
+package com.humberto.tasky.core.database.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import com.humberto.tasky.core.database.entity.EventEntity
+import com.humberto.tasky.core.database.entity.ReminderEntity
+import com.humberto.tasky.core.database.entity.TaskEntity
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface AgendaDao {
+    @Query("SELECT * FROM taskentity WHERE time BETWEEN :startOfDay AND :endOfDay")
+    fun getTasksForDay(startOfDay: Long, endOfDay: Long): Flow<List<TaskEntity>>
```

Comment:
Is this `AgendaDao` needed here? Could you not place these calls to fetch the different types of agenda items in their respective `Dao`s?